src/Raven.Server/Documents/QueueSink/Commands/BatchQueueSinkScriptCommand.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Raven.Server.Documents.Patch;
+using Raven.Server.Documents.QueueSink.Stats;
+using Raven.Server.Documents.TransactionMerger.Commands;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Logging;
+
+namespace Raven.Server.Documents.QueueSink.Commands;
+
+public sealed class BatchQueueSinkScriptCommand : DocumentMergedTransactionCommand
+{
+    private readonly List<BlittableJsonReaderObject> _messages;
+    private readonly string _script;
+    private readonly QueueSinkStatsScope _scriptProcessingScope;
+    private readonly QueueSinkProcessStatistics _statistics;
+    private readonly Logger _logger;
+
+    public BatchQueueSinkScriptCommand(string script, List<BlittableJsonReaderObject> messages, QueueSinkStatsScope scriptProcessingScope,
+        QueueSinkProcessStatistics statistics, Logger logger)
+    {
+        _messages = messages;
+        _script = script;
+        _scriptProcessingScope = scriptProcessingScope;
+        _statistics = statistics;
+        _logger = logger;
+    }
+
+    private BatchQueueSinkScriptCommand(string script, List<BlittableJsonReaderObject> messages)
+    {
+        _messages = messages;
+        _script = script;
+        _scriptProcessingScope = null;
+        _statistics = null;
+        _logger = null;
+    }
+
+    public int ProcessedSuccessfully { get; private set; }
+
+    protected override long ExecuteCmd(DocumentsOperationContext context)
+    {
+        var mainScript = new PatchRequest(_script, PatchRequestType.QueueSink);
+
+        context.DocumentDatabase.Scripts.GetScriptRunner(mainScript, false, out var documentScript);
```

Comment:
1. `readonly: false`
2. this is not returned anywhere? it returns disposable

Reply:
done